src/evaluations/data/evaluation_configs.py

```diff
@@ -498,81 +519,236 @@ def get_evaluation_config(config_name):
-    sketch_noiser=bloom_filters.BlipNoiser(epsilon=np.log(3)))
-
-BLOOM_FILTER_3E6_INFTY_UNION_ESTIMATOR = SketchEstimatorConfig(
-    name='bloom_filter-3e6-infty-union_estimator',
-    sketch_factory=bloom_filters.BloomFilter.get_sketch_factory(
-        3*10**6, 8),
-    estimator=bloom_filters.UnionEstimator())
-
-BLOOM_FILTER_3E6_LN3_UNION_ESTIMATOR = SketchEstimatorConfig(
-    name='bloom_filter-3e6-ln3-union_estimator',
-    sketch_factory=bloom_filters.BloomFilter.get_sketch_factory(
-        3*10**6, 8),
-    estimator=bloom_filters.UnionEstimator(
-        denoiser=bloom_filters.SurrealDenoiser(probability=0.25)
-    ),
-    sketch_noiser=bloom_filters.BlipNoiser(epsilon=8*np.log(3)))
-
-EXP_BLOOM_FILTER_1E5_10_LN3_FIRST_MOMENT_LOG = SketchEstimatorConfig(
-    name='exp_bloom_filter-1e5_10-ln3-first_moment_exp',
-    sketch_factory=bloom_filters.ExponentialBloomFilter.get_sketch_factory(
-        length=10**5, decay_rate=10),
-    estimator=bloom_filters.FirstMomentEstimator(
-        method=bloom_filters.FirstMomentEstimator.METHOD_EXP,
-        denoiser=bloom_filters.SurrealDenoiser(epsilon=math.log(3))),
-    sketch_noiser=bloom_filters.BlipNoiser(epsilon=math.log(3)))
-
-EXP_BLOOM_FILTER_1E5_10_LN3GLOBAL_FIRST_MOMENT_LOG = SketchEstimatorConfig(
-    name='exp_bloom_filter-1e5_10-ln3global-first_moment_exp',
-    sketch_factory=bloom_filters.ExponentialBloomFilter.get_sketch_factory(
-        length=10**5, decay_rate=10),
-    estimator=bloom_filters.FirstMomentEstimator(
-        method=bloom_filters.FirstMomentEstimator.METHOD_EXP,
-        noiser = estimator_noisers.GeometricEstimateNoiser(
-            epsilon=math.log(3))))
-
-EXP_BLOOM_FILTER_1E5_10_INFTY_FIRST_MOMENT_LOG = SketchEstimatorConfig(
-    name='exp_bloom_filter-1e5_10-infty-first_moment_exp',
-    sketch_factory=bloom_filters.ExponentialBloomFilter.get_sketch_factory(
-        length=10**5, decay_rate=10),
-    estimator=bloom_filters.FirstMomentEstimator(
-        method=bloom_filters.FirstMomentEstimator.METHOD_EXP))
+def _log_bloom_filter_first_moment_log(length, sketch_epsilon=None,
+                                       estimate_epsilon=None):
+  """Generate a SketchEstimatorConfig for Log Bloom Filters.
+
+  Args:
+    length: the length of the log bloom filters.
+    sketch_epsilon: a differential private parameter for the sketch.
+    estimate_epsilon: a differential private parameter for the estimated
+      cardinality.
+
+  Returns:
+    A SketchEstimatorConfig for log Bloom Filters of length being 10**5.
+  """
+  if sketch_epsilon:
+    sketch_noiser = bloom_filters.BlipNoiser(epsilon=sketch_epsilon)
+    sketch_denoiser = bloom_filters.SurrealDenoiser(epsilon=sketch_epsilon)
+  else:
+    sketch_noiser, sketch_denoiser = None, None
+
+  if estimate_epsilon:
+    estimate_noiser = estimator_noisers.GeometricEstimateNoiser(
+        epsilon=estimate_epsilon)
+  else:
+    estimate_noiser = None
+
+  length = int(length)
```

Comment:
why do we need this?

Reply:
To make sure that the length of the BF sketch is integer value. Otherwise, there would be error when generating the sketches. I refactored it to the ADBF_LENGTH_LIST global variables and removed this line, so that there is no need to worry about it in this function.